engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/formatter/ConfluentAvroMessageFormatter.scala

```diff
@@ -23,24 +29,59 @@ private[confluent] class ConfluentAvroMessageFormatter(schemaRegistryClient: Sch
 
   schemaRegistry = schemaRegistryClient
 
-  def writeTo(data: Array[Byte], output: PrintStream): Unit = {
-    val obj = deserialize(data)
-    val schema = AvroSchemaUtils.getSchema(obj)
+  def genericToJson(obj: AnyRef, schema: Schema): Json = {
+    toJson(obj, schema, new GenericDatumWriter[AnyRef](schema))
+  }
+
+  def specificToJson(obj: AnyRef, schema: Schema): Json = {
+    toJson(obj, schema, new SpecificDatumWriter[AnyRef](schema))
+  }
 
+  private def toJson(obj: AnyRef, schema: Schema, writer: DatumWriter[AnyRef]): Json = {
     try {
       //pretty = false is important, as we rely on the fact that there are no new lines in message parsing
+      val output = new ByteArrayOutputStream
       val encoder = encoderFactory.jsonEncoder(schema, output, false)
-      val writer = new GenericDatumWriter[AnyRef](schema)
+
       obj match {
         case bytes: Array[Byte] =>
           writer.write(ByteBuffer.wrap(bytes.asInstanceOf[Array[Byte]]), encoder)
         case _ =>
           writer.write(obj, encoder)
       }
       encoder.flush()
+      io.circe.parser.parse(output.toString).right.get
     } catch {
       case ex: AvroRuntimeException =>
         throw new SerializationException(String.format("Error serializing Avro data of schema %s to json", schema), ex)
     }
   }
+
+  def serializeRecord[K, V](record: ConsumerRecord[K, V],
+                            keySchemaIdOpt: Option[Int],
+                            keyJsonOpt: Option[Json],
+                            valueSchemaIdOpt: Option[Int],
+                            valueJsonOpt: Option[Json]
+                           ): Array[Byte] = {
+
+    val serializableRecord = SerializableConsumerRecord(
```

Comment:
Can you create factory method `SerializableConsumerRecord.apply(consumerRecord, keySchemaIdOpt, valueSchemaIdOpt)` to have this option wrapping logic in one place?

Reply:
That changes to Formatter are moved to separate PR.